### ISSUE TITLE
github action to schedule PR merge

### ISSUE THREAD

**Issue by erohmensing** [airbytehq/airbyte#24068]
E.g. using something like this [github worklow](https://github.com/airbytehq/github-workflow-test-repo-base/blob/master/.github/workflows/schedule-pr-merge.yml) - here are the github action's [docs](https://github.com/gr2m/merge-schedule-action#usage). Not married to using it if it doesn't have the functionality we are looking for!

Although there is the following parameter:
```
          # Require all pull request statuses to be successful before
          # merging. Default is `false`.
          require_statuses_success: 'true'
```
I noticed that it [still merged](https://github.com/airbytehq/github-workflow-test-repo-base/pull/94) even when I set `A hypothetical required test` to being required under a branch protection rule, which is unacceptable and needs digging into why that happened.

The goal is for TCS to be able to schedule changes to a connector for a certain date without having to babysit all of those PRs and coordinate their merge on the specified date. We'd like to set it and forget it.

**Comment by erohmensing** [comment]
Grooming
* there is a bit more to this - we don't want to publish until right before merge 
  * This will be made easy when we update the workflow to publish on merge instead of with the `/publish` command

**Comment by erohmensing** [comment]
Grooming: 

For connector changes, a lot of this workflow seems to be blocked on making sure CAT passes and publish is only run right before merge
However, we can still work on the workflow part of this, e.g. for platform changes now that CD is back in action

Goals:
* An action that can merge a PR on a given day/other schedule
* The action fails + notifies user if required tests are not passing
* Probably out of scope: we encourage devs to make sure _everything_ is green, not just required checks
*

**Comment by evantahler** [comment]
Moving to the icebox, because hopefully we won't need this as we get to work on `Automatic Management of Connector Breaking Change Releases`

**Comment by bnchrch** [comment]
Epic grooming:
1. close this in favor of breaking change rollout?

**Comment by evantahler** [comment]
I vote that we can close this issue, but I think it's @erohmensing
